src/content/docs/en/reference/container-reference.mdx

```diff
@@ -86,19 +86,24 @@ const result = await container.renderToString(ReactWrapper);
 
 #### Adding a renderer manually
 
-When the container is called at runtime, or inside other "shells", the `astro:container` virtual module's helper functions are not available. Import the necessary server renderer manually and store it inside the container using `experimental_Astro.Container.addServerRenderer`.
+When the container is called at runtime, or inside other "shells", the `astro:container` virtual module's helper functions are not available.  You must import the necessary server and client renderers manually and store them inside the container using `addServerRenderer` and `addClientRenderer`.
 
-The following example manually imports the server renderers for React and Vue components, and MDX for `.mdx` pages. 
+Server renderers are required to build your project, and must be stored in the container for every framework used. Client renderers are additionally needed to any hydrate client-side components using [`client:*` directives](/en/reference/directives-reference/#client-directives).
+
```

Comment:
```suggestion
Server renderers are required to build your project, and must be stored in the container for every framework used. Client renderers are additionally needed to any hydrate client-side components using [`client:*` directives](/en/reference/directives-reference/#client-directives).

Only one import statement is needed per framework. Importing a renderer makes both the server and client renderers available to your container. However, **server renderers must be added to your container before client renderers**. This allows your entire container to build first, and then hydrate any interactive components.
```
Instead of a note, does this accurately describe what's happening? A paragraph here that both mentions the order requirement and explains why might make the information more "sticky" for people, since this parallels how they think about Astro sites in general.

If this is the case, I might adjust the blank line in the example below to group all the server renderers together, and leave the client renderer on its own after the blank line. This would be another way to subtly enforce this pattern. 

(I don't actually know whether ALL the server renders need to be imported first, before ANY client renderers, or whether it's just to make sure for each framework, the client comes second. But if we show importing ALL the server renderers first, people will get the idea.)

Reply:
@ematipico If you're OK with the above change, (or just edit it slightly if it needs correcting) then I think this can merge any time!

Reply:
Thank you! I applied the edits, and I'm going to merge